pkg/api/frontendsettings.go

```diff
@@ -371,6 +371,16 @@ func (hs *HTTPServer) getFSDataSources(c *contextmodel.ReqContext, availablePlug
 			}
 		}
 
+		// Update `jsonData.database` for outdated provisioned SQL datasources created WITHOUT the `jsonData` object in their configuration.
+		// In these cases, the `Database` value is defined (if at all) on the root level of the provisioning config object.
+		// This is done for easier warning/error checking on the front end.
+		if (ds.Type == datasources.DS_MSSQL) || (ds.Type == datasources.DS_MYSQL) || (ds.Type == datasources.DS_POSTGRES) {
```

Comment:
I'd also add the feature flag check to this bit of code too. WDYT?

Reply:
I actually don't think we need a FF here; its a piece of code that makes using outdated provisioned SQL databases easier to work with on the FE, and should be added regardless IMO. @zoltanbedi any thoughts on this one?

Reply:
I can get behind that, sounds good!

Reply:
Wouldn't do any harm to have this without a feature flag.